src/views/Update/components/UpdatePage/index.js

```diff
@@ -0,0 +1,365 @@
+        return false;
+      }
+    }
+    return true;
+  }, [updatedInfo, currentInfo]);
+
+  // handles all field changes
+  const handleChange = (event) => {
+    const getNewInfo = (currentValue) => {
+      return {
+        ...currentValue,
+        [event.target.name]:
+          event.target.type === 'checkbox' ? event.target.checked : event.target.value,
+      };
+    };
+    setUpdatedInfo(getNewInfo);
+  };
+
+  const createSnackbar = (message, severity) => {
+    setSnackbar({ message: message, severity: severity, open: true });
+  };
+
+  // returns label given name of field
+  const getAssociatedLabel = (fieldName) => {
+    let label = '';
+    allFields.forEach((field) => {
+      if (fieldName === field.name) label = field.label;
+    });
+    return label;
+  };
+
+  /**
+   * @param updatedInfo updated information fields object
+   * @param currentInfo old/saved information fields object
+   * @returns {Array<Object>} Array of updated fields in a subobject
+   */
+  function getUpdatedFields(updatedInfo, currentInfo) {
+    var updatedFields = [];
+    for (const field in currentInfo) {
+      if (updatedInfo[field] !== currentInfo[field]) {
+        updatedFields.push({
+          field: field,
+          value: currentInfo[field],
+          label: getAssociatedLabel(field),
+        });
+      }
+    }
+    return updatedFields;
+  }
+
+  //handles confirmation window's final submit
+  const handleConfirm = () => {
+    setSaving(true);
+    let updateRequest = getUpdatedFields(currentInfo, updatedInfo);
+    let updateRequestEmail = getUpdatedFields(currentInfo, updatedInfo);
+    updateRequestEmail.push({
+      field: 'changeReason',
+      value: changeReason,
+      label: 'Reason for change',
+    });
+    requestInfoUpdate(updateRequestEmail, updateRequest).then(() => {
+      createSnackbar(
+        'A request to update your information has been sent. Please check back later.',
+        'info',
+      );
+      setSaving(false);
+      handleWindowClose();
+    });
+  };
+
+  //executes on confirmation window cancel & submit
+  const handleWindowClose = () => {
+    setOpenConfirmWindow(false);
+    setChangeReason('');
+    setConfirmRows('');
+  };
+
+  const handleSaveButtonClick = () => {
+    if (updatedInfo.firstName === '' || updatedInfo.lastName === '') {
+      createSnackbar('Please fill in your first and last name.', 'error');
+    } else {
+      getCurrentChanges(currentInfo, updatedInfo);
+      setOpenConfirmWindow(true);
+    }
+  };
+
+  const saveButton = isSaving ? (
+    <GordonLoader size={32} />
+  ) : (
+    <Button
+      variant="contained"
+      color="secondary"
+      onClick={handleSaveButtonClick}
+      disabled={hasNoChange}
+    >
+      Update
+    </Button>
+  );
+
+  /**
+   * @param {Array<Object>} fields static fields that define label, name, value, type
+   * @returns JSX elements of all elements in array with correct type
+   */
+  const infoMap = (fields) => {
```

Comment:
This comment can be much clearer. Also, please rename this function as its current name is too vague to be helpful.
```suggestion
  /**
   * @param {Array<{name: string, label: string, type: string, menuItems: string[]}>} fields array of objects defining the properties of the input field
   * @returns JSX input for each field
   */
  const mapFieldsToInputComponents = (fields) => {
```